sdk/storage/Azure.Storage.Files.DataLake/samples/Sample02_Auth.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Storage;
+using Azure.Storage.Files.DataLake;
+using Azure.Storage.Files.DataLake.Models;
+using Azure.Storage.Files.DataLake.Sas;
+using Azure.Storage.Sas;
+using NUnit.Framework;
+
+namespace Azure.Storage.Files.DataLake.Samples
+{
+    /// <summary>
+    /// Demonstrate various authorization and authentication mechanisms.
+    ///
+    /// For more information, see
+    /// https://docs.microsoft.com/en-us/azure/storage/common/storage-auth
+    /// </summary>
+    public class Sample02_Auth : SampleTest
+    {
+        /// <summary>
+        /// Anonymously access a public DataLake File.
+        ///
+        /// You can enable anonymous, public read access to a container and its
+        /// blobs in Azure Blob storage.  By doing so, you can grant read-only
+        /// access to these resources without sharing your account key, and
+        /// without requiring a shared access signature (SAS).
+        ///
+        /// Public read access is best for scenarios where you want certain
+        /// blobs to always be available for anonymous read access.  For more
+        /// fine-grained control, you can create a shared access signature.
+        /// </summary>
+        [Test]
+        public async Task AnonymousAuthAsync()
+        {
+            // Make StorageSharedKeyCredential to pass to the serviceClient
+            string accountName = StorageAccountName;
+            string accountKey = StorageAccountKey;
+            Uri serviceUri = StorageAccountBlobUri;
+            StorageSharedKeyCredential sharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
+
+            // Get a reference to a service Client
+            DataLakeServiceClient service = new DataLakeServiceClient(serviceUri, sharedKeyCredential);
+
+            // Get a reference to a filesystem named "sample-filesystem"
+            DataLakeFileSystemClient filesystem = service.GetFileSystemClient(Randomize("sample-filesystem"));
+            try
+            {
+                // Create a file that can be accessed publicly
+                await filesystem.CreateAsync(PublicAccessType.Container);
```

Comment:
Is there a bug on renaming `Container` to `FileSystem` here?  If not can we file one?

Reply:
Made an issue for this #8585